ParkingLot/src/main/java/services/ExitGate.java

```diff
@@ -1,28 +0,0 @@
-package services;
-
-import dao.TicketsDao;
-import factories.CostComputationFactory;
-import factories.ParkingSpotManagerFactory;
-import models.Ticket;
-
-public class ExitGate {
-    private final ParkingSpotManagerFactory parkingSpotManagerFactory;
-    private final CostComputationFactory costComputationFactory;
```

Comment:
Also, cost computation?

Reply:
This is created to support different ways to compute cost. For instance, some use case might want to calculate hourly other might have a flat cost structure.

Reply:
I meant why is it in exitgate, I can not create an exit gate without costComputationFactory

Reply:
ideally exitgate would have one function, markExit() in which, the driver would show the ticket(which means an argument) and ticket has the all the info(Vehicle, in time , out time) based on that, it can itself decide which cost computation to use

Reply:
We are following this approach only. This is a factory, the actual instance is fetched inside the calcaulteCost method by using the correct computation class based on the details shared via the ticket. 


If we convert the costComputationFactory.getCostComputationByVehicleType method to static, we won't need this instance here and it would look exactly the way you described.